modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -1864,6 +1864,7 @@ def apply_full_axis(
         new_index=None,
         new_columns=None,
         dtypes=None,
+        func_may_change_complementary_index_size=True,
```

Comment:
What if we would pass new row lengths and new column widths from the QC?

Reply:
The QC probably shouldn't know anything about this thought. However, would do you thing on this?

cc @mvashishtha, @devin-petersohn